evalml/tests/automl_tests/test_automl.py

```diff
@@ -1942,12 +1992,24 @@ class Pipeline2(DummyPipeline):
             y_train=y,
             problem_type=problem_type_value,
             max_iterations=3,
-            allowed_pipelines=[Pipeline1({}), Pipeline2({})],
             objective=objective,
             additional_objectives=[],
             optimize_thresholds=False,
             n_jobs=1,
         )
+    automl._automl_algorithm = IterativeAlgorithm(
```

Comment:
Did you need to define this since we're patching IterativeAlgo? Why did we not need to do it before?